packages/@atjson/offset-annotations/src/annotations/indent.ts

```diff
@@ -0,0 +1,8 @@
+import { InlineAnnotation } from "@atjson/document";
+
+export default class Indent extends InlineAnnotation<{
+  size: number;
```

Comment:
In general, this is a reasonable model for indents. However, in the specific use case of structured Poetry (which, for sake of context, is where we are wanting to use this) it makes sense to actually include leading spaces within the text, as that spacing should be considered part of the content, in my opinion. This also makes the content more friendly for copy+paste, where those space characters allow the spacing to preserved in a reasonable way. There are other domains where we probably don't want that behavior for indents, but I think that makes sense in Poetry.

If we go that route, then we'd have some leading space characters and this Indent annotation would cover those spaces, and in that case the `size` should be derived from the number of space characters rather than be an attribute on this annotation.

Reply:
Hi @bachbui , thanks for your comment, however I don't follow your train of thought entirely.

Where should those spaces be added? Are you suggesting this annotation would include the text with spaces? Or only a string inside which are all spaces, to be later pre-pended to the text?

BTW The idea behind that size attribute is to later on render `&emsp;` spaces, eg:

`size: 3` => `&emsp;&emsp;&emsp;indented text`

`&emsp;` spaces are copy-pasteable. Also, if this indent would later render 'normal' spaces, the actual width of each space and hence the number of spaces required is less deterministic, as opposed to using '&emsp'

cc @dcolucci

Reply:
Hi @gnorsilva! The tldr is I am proposing that for the fixed-size indents we are trying to describe here, we do not need a size attribute, because these annotations should cover some number of leading spaces in the content. 

This may be confusing since there are several domains we are talking about, and these are modeled slightly differently in each domain. Here is a lengthier explanation:

In some extended markdown, we have encountered fixed-sized indents as the following *markup string*:
`*some text*{: .indentN}`

As an AtJSON Markdown document, we'd model this as:
```
{
  content: 'some text',
  annotations: [{
    type: '-commonmark-emphasis',
    start: 0,
    end: 9,
    attributes: {
      class: 'indentN'
    }
  }]
}
```

We know we want to transform this extended markdown into a markdown that is more widely supported. In particular, we want to produce the *markdown string*
```
&emsp;...&emsp;some text
```

The question of this PR is what does the AtJSON document look like in a common format, here in `OffsetAnnotations`, in order to render this markdown? What I am proposing is that document looks like
```
{
  content: '\u2003...\u2003some text',
  annotations: [{
    type: '-offset-fixed-indent',
    start: 0,
    end: N
  }]
}
```
In the conversion from the markdown AtJSON document to the Offset AtJSON document, note we would insert leading space characters to the content string (\u2003 is the unicode character for an emspace), and we have added a FixedIndent annotation covering these leading spaces. Because of the presence of leading spaces in this document, we do not need a size attribute on the FixedIndent annotation, since it can be derived from the text that it covers.

The advantage of modeling like this is that if we were to use a renderer that does not have been updated to know how to render a FixedIndent annotation, we'd still get (in most cases) a reasonable result, since the spacing is encoded as part of the text. In particular, to render the desired markdown, we might not have to do *any* special handling.

Reply:
I will also add that if we go with the above, then we probably want distinct annotations for what I've called a 'FixedIndent' above (one suitable for structured poetry, and expressed in text) from a regular 'Indent' (one which is more of a layout control, which would not be expressed in text)

Reply:
Thanks for clarifying @bachbui :)

I know very little of the atjson repo, my gut feeling would be to not alter the actual text content but rather have it as an indent annotation, but I'll give this FixedIndent approach a shot

Reply:
Sure thing, let me know if you have any questions. I'm going to tag @tim-evans @colinarobinson and @blaine here as well in case they have additional thoughts (or think what I'm suggesting is way off 😄 )

Reply:
I'm glad I held off, because I'd misunderstood what Bach was saying earlier in the thread, and the clarification above is _exactly_ what I'd imagine (tbh, even better than the specific implementation I'd imagined).